system/behavior.lua

```diff
@@ -70,29 +71,44 @@ function Behavior:Update()
   -- if no behavior is active, return
   if not behavior then return end
 
-  -- Call all behaviors in whatever order they are in
-  -- Could sort to call them in a specific order
-  for _, k in pairs(BehaviorType) do
-    if behavior.Behaviors[k] then
-      behavior.Behaviors[k]()
+  local className = Me.ClassName:lower():gsub("%s+", "")
+  local specname = Me:SpecializationName()
+
+  -- Loop through BehaviorTypes
+  for _, type in pairs(BehaviorType) do
+    -- Check if a callback exists for the player's class, specialization and behavior type
+    if behavior.Callbacks[className] and behavior.Callbacks[className][specname] and behavior.Callbacks[className][specname].Behaviors and behavior.Callbacks[className][specname].Behaviors[type] then
+      -- Call the callback
+      behavior.Callbacks[className][specname].Behaviors[type]()
     end
-    for _, v in pairs(self.Extras) do
-      if v.Behaviors[k] then
-        v.Behaviors[k]()
-      end
+  end
+
+  -- Run Extras separately
+  for _, extraBehavior in pairs(self.Extras) do
+    if extraBehavior.Behaviors[BehaviorType.Extra] then
+      extraBehavior.Behaviors[BehaviorType.Extra]()
     end
   end
 end
-
 function Behavior:setActive(behavior)
   print(string.format('Setting active behavior to %s', behavior.Name))
   Settings.ActiveBehavior = behavior.Name
   self.Active = behavior
 
-  -- XXX: add back when we can change ImText text value
+  -- find the appropriate options for this specialization
+  local className = Me.ClassName:lower():gsub("%s+", "")
+  local specname = Me:SpecializationName()
+
+  if behavior.Callbacks[className] and behavior.Callbacks[className][specname] and behavior.Callbacks[className][specname].Options then
```

Comment:
Remove this, it will add a new option sub menu anytime you select another behavior which means duplication until reload.